sympy/solvers/ode/single.py

```diff
@@ -359,6 +358,137 @@ def inverse(self):
+        if self.ode_problem.order!=1:
+            self._matched=False
+            return False
+
+        eq= self.ode_problem.eq
+        f = self.ode_problem.func
+        x = self.ode_problem.sym
+        y=Dummy('y')
+
+        self.q=collect(eq,f.diff(x)).coeff(f.diff(x))
+        self.p=eq-self.q*f.diff(x)
+
+        try:
+            if simplify(self.p)!=0:
+                m=self.p.subs(f,y)
+                n=self.q.subs(f,y)
+
+                numerator= simplify(m.diff(y)-n.diff(x))
+                # The following few conditions try to convert a non-exact
+                # differential equation into an exact one.
+                # References :
+                #1. Differential equations with applications
+                # and historical notes - George E. Simmons
+                #2. https://math.okstate.edu/people/binegar/2233-S99/2233-l12.pdf
+                if numerator:
+                    # If (dP/dy - dQ/dx) / Q = f(x)
+                    # then exp(integral(f(x))*equation becomes exact
+                    factor = simplify(numerator/n)
+                    variables = factor.free_symbols
+
+                    if len(variables) == 1 and x == variables.pop():
+                        factor = exp(Integral(factor).doit())
+                        m*= factor
+                        n*= factor
+                        self.p=m.subs(y,f)
+                        self.q=n.subs(y,f)
+                        self._matched=True
+                        return True
+                    else:
+                        # If (dP/dy - dQ/dx) / -P = f(y)
+                        # then exp(integral(f(y))*equation becomes exact
+                        factor = simplify(-numerator/m)
+                        variables = factor.free_symbols
+                        if len(variables) == 1 and y == variables.pop():
+                            factor = exp(Integral(factor).doit())
+                            m*= factor
+                            n*= factor
+                            self.p=m.subs(y,f)
+                            self.q=n.subs(y,f)
+                            self._matched=True
+                            return True
+                else:
+                    self._matched=True
+                    return True
+        except NotImplementedError:
+            # Differentiating the coefficients might fail because of things
+            # like f(2*x).diff(x).  See issue 4624 and issue 4719.
+            pass
+
+        self._matched=False
+        return False
+
+    def _get_general_solution(self, *, simplify: bool = True) -> List[Equality]:
+        y=Dummy('y')
+        f=self.ode_problem.func
+        x=self.ode_problem.sym
+        (C1,) = self.ode_problem.get_numbered_constants(num=1)
+
+        m=self.p.subs(f,y)
+        n=self.q.subs(f,y)
+
+        sol= Integral(m,x)+Integral(n-Integral(m,x).diff(y),y)
+
+        try:
+            sol=sol.doit()
+            gen_sol=Eq(sol.subs(y,f),C1)
```

Comment:
We don't need to call `doit` here as it will be called by dsolve later.

Reply:
Updating it.  As there is no other way to pass `y` to `_handle_Integral`, global `y` hack will still be in use.